### PR TITLE
chore: fix typos

### PR DESCRIPTION
**Pull request by realeinherjar**
### Description

Fixes the typos and remove unused speculos dockerfiles that was done in #1165.
Moving these changes into this PR to be merged first.
Then, we can rebase #1165 and make it only related to CI and Nix.
(Maybe do a big squash 😄)

## Note to Reviewers

About the speculos stuff, we are not using them, removed in #793,
more specifically in 3f5a78ae3b1e6c3f0a6acc98bb2445c895cfd743.

### Changelog notice

- Fix typos in codebase and docs
- Remove unused CI tests with hardware signer Dockerfiles

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [ ] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

### CODE REVIEW

**Comment by realeinherjar** [comment]
Cc @danielabrozzoni

File: ci/Dockerfile.ledger [hunk 1]
```diff
@@ -1,9 +0,0 @@
-# Taken from bitcoindevkit/rust-hwi
```
Comment (danielabrozzoni) [inline ci/Dockerfile.ledger]: Can we avoid removing this dockerfile and speculos' `automation.json`? I'm doing some changes to hwi in #1161, so I will need them again

File: .github/workflows/code_coverage.yml [hunk 2]
```diff
@@ -27,16 +27,6 @@ jobs:
         uses: Swatinem/rust-cache@v2.2.1
       - name: Install grcov
         run: if [[ ! -e ~/.cargo/bin/grcov ]]; then cargo install grcov; fi
-      - name: Build simulator image
```
Comment (danielabrozzoni) [inline .github/workflows/code_coverage.yml:30]: Can we just comment this one instead of deleting it? With a `TODO: re-enable the hwi tests`

**Review by danielabrozzoni — COMMENTED** [review]
Looks good to me, a couple of comments

File: ci/Dockerfile.ledger (hunk 1 above)
Comment (realeinherjar) [reply ci/Dockerfile.ledger]: Done

File: .github/workflows/code_coverage.yml (hunk 2 above)
Comment (realeinherjar) [reply .github/workflows/code_coverage.yml:30]: Done! Glad that you are working on re-enabling that.

**Review by danielabrozzoni — APPROVED** [review]
ACK 27a63abd1e7c87aaf3e07f8023370887a270c61a

**Pull request merged**